lang/cs/Org.Apache.REEF.Common/Telemetry/MetricsService.cs

```diff
@@ -65,13 +89,64 @@ public void OnNext(IContextMessage contextMessage)
                     //// For the counters from multiple evaluators with the same counter name, the value should be aggregated here
                     //// We also need to consider failure cases.  
                     _counters[counter.Name] = counter;
+
+                    if (_totalIncrementSinceLastSink.ContainsKey(counter.Name))
+                    {
+                        _totalIncrementSinceLastSink[counter.Name] += counter.Value - c.Value;
+                    }
+                    else
+                    {
+                        _totalIncrementSinceLastSink.Add(counter.Name, counter.Value - c.Value);
+                    }
                 }
                 else
                 {
                     _counters.Add(counter.Name, counter);
+                    _totalIncrementSinceLastSink.Add(counter.Name, counter.Value);
                 }
 
-                Logger.Log(Level.Verbose, "Counter name: {0}, value: {1}, description: {2}, time: {3}.", counter.Name, counter.Value, counter.Description, new DateTime(counter.Timestamp));
+                Logger.Log(Level.Verbose, "Counter name: {0}, value: {1}, description: {2}, time: {3},  incrementSinceLastSink: {4}.", 
+                    counter.Name, counter.Value, counter.Description, new DateTime(counter.Timestamp), _totalIncrementSinceLastSink[counter.Name]);
+            }
+
+            if (TriggerSink())
+            {
+                SinkCounters();
+                _totalIncrementSinceLastSink.Clear();
+            }
+        }
+
+        /// <summary>
+        /// The condition that triggers the sink. The condition can be modified later.
```

Comment:
per offline conversation - this will be for next iteration

---
In reply to: [110776972](https://github.com/apache/reef/pull/1262#discussion_r110776972) [](ancestors = 110776972)